src/Microsoft.Azure.WebJobs.Host/Common/ITriggerBindingStrategy.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Strategy pattern to describe how to bind a core trigger type to various parameter.
+    /// Suports:
+    /// - both Single-item vs. Batch dispatch.  
+    /// - core binding, string, poco w/ Binding Contracts
+    /// 
+    /// For example, a single EventHubTriggerInput -->  can bind to 
+    ///  EventData, EventData[], string, string[], Poco, Poco[]    
+    /// </summary>
+    /// <typeparam name="TMessage">The native message type. For Azure Queues, this would be CloudQueueMessage.</typeparam>
+    /// <typeparam name="TTriggerValue">The type of the trigger object that the listener returns. This could represent a single message or a batch of messages.</typeparam>
+    public interface ITriggerBindingStrategy<TMessage, TTriggerValue>
```

Comment:
agreed.

---
In reply to: [54937056](https://github.com/Azure/azure-webjobs-sdk/pull/643#discussion_r54937056) [](ancestors = 54937056)